build.py

```
@@ -286,15 +316,28 @@ def clone_or_update_repo(name, branch, dir_name, owner="openwisp", dest=None):
     if os.path.exists(clone_path):
         print(f"Repository '{name}' already exists. Updating...")
         subprocess.run(
-            # Clears ambiguity when git tags and branches have identical names
-            ["git", "fetch", "origin", f"refs/heads/{branch}:refs/heads/{branch}"],
+            # Update remote-tracking ref (avoid fetching into checked-out branch)
+            [
+                "git",
+                "fetch",
+                "origin",
+                f"+refs/heads/{branch}:refs/remotes/origin/{branch}",
+            ],
```

Ruff (0.14.14)

Starting a process with a partial executable path

(S607)